src/App.tsx

```diff
@@ -1,14 +1,67 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { actions as todosActions } from './features/todos';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [isLoader, setIsLoader] = useState<boolean>(false);
+
+  const dispatch = useAppDispatch();
+  const todos = useAppSelector(state => state.todos);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const { status, query } = useAppSelector(state => state.filter);
+
+  const lowerQuery = query.toLowerCase();
+
+  const getData = async () => {
+    try {
+      setIsLoader(true);
+      let loadedData = await getTodos();
+
+      switch (status) {
+        case 'all':
+          break;
+
+        case 'active':
+          loadedData = [...loadedData].filter((data) => !data.completed);
+          break;
+
+        case 'completed':
+          loadedData = [...loadedData].filter((data) => data.completed);
+          break;
+
+        default:
+          break;
+      }
+
+      if (query.length) {
+        loadedData = [...loadedData].filter((data) => {
+          const lowerTitle = data.title.toLowerCase();
+
+          return lowerTitle.includes(lowerQuery);
+        });
+      }
+
+      dispatch(todosActions.setTodos(loadedData));
+    } catch (error) {
+      throw Error('Field to load todo data');
+    } finally {
+      setIsLoader(false);
+    }
+  };
+
+  useEffect(() => {
+    getData();
```

Comment:
In your app every re-render you make a new request to the server to get data. It is an expensive approach.
It is better to load data once the app is loaded for the first time, set them to store, and then just filter them in your app depending on filter options. And then just render those `visibleTodos`

Try to implement this